src/primitives/torus.jl

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Torus(majorRadius, minorRadius)
+
+A torus with `majorRadius` and `minorRadius` with the z axis as its axis of 
+rotation and the xy plane as its plane of reflection. 
+
+"""
+struct Torus{T} <: Primitive{3,T}
+  majorRadius::T
+  minorRadius::T
+end
+
+function Torus(majorRadius::T1, minorRadius::T2) where {T1,T2}
+  T = promote(T1, T2)  
```

Comment:
That is not how you use promote, the correct usage is `r1, r2 = promote(major, minor)`. Are you running these codes before submitting a PR for review?

Reply:
Not when I'm sure of me :-D I should not be so self-confident sometimes.